### ISSUE TITLE
`no-unmodified-loop-condition` false positive

### ISSUE THREAD

**Issue by feross** [eslint/eslint#5166]
Using eslint 2.0.0-rc.0.

Code:

```js
var i = 0
var j = 10
while (++i < j) {
  console.log(i)
}
```

Rules:

```json
{
  "rules": {
    "no-unmodified-loop-condition": 2
  }
}
```

I get this unexpected error:

```
3:14  error  'j' is not modified in this loop  no-unmodified-loop-condition
```

The examples in the docs indicate that this rule is smart enough to handle cases where "the result of a binary expression is changed in the loop". Seems there's a bug.

**Comment by eslintbot** [comment]
@feross Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by feross** [comment]
The docs indicate that this detection works fine for for-loops, for instance:

```js
for (var j = 0; j < items.length; ++j) {
    doSomething(items[j]);
}
```

Where it detects that `j` is modified here.

**Comment by ilyavolodin** [comment]
Well in the example from the documentation `j` is actually modified. In your case, only `i` is modified, but `j` stays static. So your code can be rewritten as:
```js
var i = 0
while (++i < 10) {
  console.log(i)
}
```
I think the rule is doing the right thing in this case.

Although `j` is not used inside the loop, so I'm not sure if this is correct behavior. @mysticatea any insights?

**Comment by feross** [comment]
Right. But it seems like the *intent* of the rule, based on the docs and even the rule name, was to detect unchanged *conditions*. Not to check that every variable in the condition is modified.

**Comment by mysticatea** [comment]
Thank you for good catch!
Yes, this is a bug. I will fix this.

**Comment by IanVS** [comment]
@ilyavolodin I agree this looks like a bug.  Your counter example of how the code could be rewritten introduces a magic number (10), and I agree with @feross that the intent is just to make sure _something_ in the condition is changing.

**Comment by jakepusateri** [comment]
```javascript
var x = 100, y, offset = 12;
for (y = 100; y > (x - offset); y--) {
    // ...
}
```

The above example produces the following errors:

    error  'x' is not modified in this loop       no-unmodified-loop-condition
    error  'offset' is not modified in this loop  no-unmodified-loop-condition

`y` is changing in this loop, so `x` and `offset` should be allowed to be constants, correct?

**Comment by feross** [comment]
@jakepusateri Are you sure you're using the latest version of eslint? The code you pasted works fine for me.
